rm-empty/commands/folders.go

```diff
@@ -0,0 +1,198 @@
+		return errors.New("Wrong number of arguments. Expected: 1, " + "Received: " + strconv.Itoa(len(c.Arguments)))
+	}
+
+	rtDetails, err := getRtDetails(c)
+	if err != nil {
+		return err
+	}
+
+	path := c.Arguments[0]
+	return deleteEmptyFolders(rtDetails, path, c.GetBoolFlagValue("quiet"))
+}
+
+// Deletes all the empty folders under the specified path in Artifactory.
+func deleteEmptyFolders(rtDetails *config.ServerDetails, path string, quiet bool) (err error) {
+	// Create a search command, to find all the files and folders under the specified path.
+	spec := spec.NewBuilder().Pattern(path).IncludeDirs(true).Recursive(true).BuildSpec()
+	cmd := generic.NewSearchCommand()
+	cmd.SetServerDetails(rtDetails).SetSpec(spec).SetRetries(3)
+
+	log.Info("Searching for all items under", path)
+
+	// Run the search and receive a reader with the results.
+	var reader *content.ContentReader
+	reader, err = cmd.Search()
+	if err != nil {
+		return
+	}
+	defer func() {
+		e := reader.Close()
+		if err == nil {
+			err = e
+		}
+	}()
+
+	// Sort the results in the reader, so that the empty folders can be found by reading the results
+	// record by record.
+	var sortedFilesReader *content.ContentReader
+	sortedFilesReader, err = content.SortContentReader(clientrtutils.ResultItem{}, reader, true)
+	if err != nil {
+		return
+	}
+
+	// Create a writer, that will be used to store the paths of the empty folders found.
+	var emptyFoldersWriter *content.ContentWriter
+	emptyFoldersWriter, err = content.NewContentWriter(content.DefaultKey, true, false)
+	defer func() {
+		e := emptyFoldersWriter.Close()
+		if err == nil {
+			err = e
+		}
+	}()
+
+	// Find all empty folders by scanning the sortedFilesReader, and write them into the emptyFoldersWriter.
+	filterEmptyFolders(sortedFilesReader, emptyFoldersWriter)
+
+	// Create a reader from the writer, to read all the empty folders found.
+	emptyFoldersReader := content.NewContentReader(emptyFoldersWriter.GetFilePath(), content.DefaultKey)
+	defer func() {
+		e := emptyFoldersReader.Close()
+		if err == nil {
+			err = e
+		}
+	}()
+
+	var length int
+	length, err = emptyFoldersReader.Length()
+	if err != nil {
+		return
+	}
+	if length == 0 {
+		log.Info("No empty folders found")
+		return
+	}
+
+	// Delete the folders in the reader.
+	deleteItem(emptyFoldersReader, rtDetails, quiet)
+	return
+}
+
+// Find all empty folders by scanning the sortedFilesReader, and write them into the emptyFoldersWriter.
+func filterEmptyFolders(sortedFilesReader *content.ContentReader, emptyFoldersWriter *content.ContentWriter) {
+	var prevFolder *clientrtutils.ResultItem
+	for item := new(clientrtutils.ResultItem); sortedFilesReader.NextRecord(item) == nil; item = new(clientrtutils.ResultItem) {
+		if prevFolder != nil && !strings.HasPrefix(item.Path, prevFolder.Path) {
+			emptyFoldersWriter.Write(prevFolder)
+		}
+		if item.Type == "folder" {
+			prevFolder = item
+		} else {
+			prevFolder = nil
+		}
+	}
+	if prevFolder != nil {
```

Comment:
If a repository is empty, this will return true. I think we should take into account the empty repository scenario